homeassistant/components/tuneblade_remote/config_flow.py

```diff
@@ -0,0 +1,130 @@
+    """Config flow for TuneBlade Remote."""
+
+    VERSION = 1
+
+    def __init__(self) -> None:
+        """Initialize."""
+        self._discovery_info: dict[str, Any] = {}
+        self._title_placeholders: dict[str, str] = {}
+
+    async def async_step_user(
+        self, user_input: dict[str, Any] | None = None
+    ) -> ConfigFlowResult:
+        """Manual user setup step."""
+        errors = {}
+
+        if user_input is not None:
+            name = user_input["name"].strip()
+            host = user_input["host"]
+            port = user_input["port"]
+
+            session = async_get_clientsession(self.hass)
+            client = TuneBladeApiClient(host, port, session)
+            try:
+                devices = await client.async_get_data()
+            except Exception:
+                _LOGGER.exception("Failed to connect to TuneBlade")
+                errors["base"] = "cannot_connect"
+            else:
+                if not devices:
+                    errors["base"] = "cannot_connect"
+
+            if not errors:
+                unique_id = f"{name}_{host}_{port}"
+                await self.async_set_unique_id(unique_id)
+                self._abort_if_unique_id_configured()
+
+                return self.async_create_entry(
+                    title=name,
+                    data={
+                        "host": host,
+                        "port": port,
+                        "name": name,
+                    },
+                )
+
+        data_schema = vol.Schema(
+            {
+                vol.Required("host", default="localhost"): str,
+                vol.Required("port", default=54412): int,
+                vol.Optional("name", default="TuneBlade"): str,
+            }
+        )
+        return self.async_show_form(
+            step_id="user", data_schema=data_schema, errors=errors
+        )
+
+    async def async_step_zeroconf(
+        self, discovery_info: ZeroconfServiceInfo
+    ) -> ConfigFlowResult:
+        """Handle zeroconf discovery."""
+        _LOGGER.debug("Discovered TuneBlade via Zeroconf: %s", discovery_info)
+
+        host = discovery_info.host
+        port = discovery_info.port
+        name = discovery_info.name.split("@")[0].strip()
+
+        unique_id = f"{name}_{host}_{port}"
+        await self.async_set_unique_id(unique_id)
+        self._abort_if_unique_id_configured()
+
+        self._discovery_info = {
+            "host": host,
+            "port": port,
+            "name": name,
+        }
+
+        session = async_get_clientsession(self.hass)
+        client = TuneBladeApiClient(host, port, session)
+        try:
+            devices = await client.async_get_data()
+        except Exception:
+            _LOGGER.exception("Failed to connect to TuneBlade via zeroconf")
+            return self.async_abort(reason="cannot_connect")
+
+        if not devices:
+            return self.async_abort(reason="cannot_connect")
+
```

Comment:
why would there be no data

Reply:
I've updated this so it doesn't abort. If TuneBlade hasn't discovered any AirPlay devices on the network then it's possible this would be empty. Not sure why you'd use it in this case but perhaps a speaker has gone off line